qiskit/compiler/assemble.py

```diff
@@ -289,20 +293,25 @@ def _parse_pulse_args(backend, qubit_lo_freq, meas_lo_freq, qubit_lo_range,
     if rep_time:
         if dynamic_reprate_enabled:
             warnings.warn("Dynamic rep rates are supported on this backend. 'rep_delay' will be "
-                          "used instead, if specified.", RuntimeWarning)
+                          "used instead of 'rep_time'.", RuntimeWarning)
         if isinstance(rep_time, list):
             rep_time = rep_time[0]
         rep_time = int(rep_time * 1e6)  # convert sec to μs
 
-    rep_delay = rep_delay or getattr(backend_config, 'rep_delays', None)
-    if rep_delay:
-        if not dynamic_reprate_enabled:
-            warnings.warn("Dynamic rep rates not supported on this backend. 'rep_time' will be "
-                          "used instead.", RuntimeWarning)
+    if dynamic_reprate_enabled:
+        rep_delay = rep_delay or getattr(backend_config, 'default_rep_delay', None)
+        if rep_delay:
+            rep_delay_range = getattr(backend_config, 'rep_delay_range', [-1, -1])
+            # check that rep_delay is in rep_delay_range
+            if rep_delay and not rep_delay_range[0] <= rep_delay <= rep_delay_range[1]:
+                raise QiskitError("Supplied rep delay {} not in the allowed "
+                                  "backend rep delay range {}".format(rep_delay, rep_delay_range))
 
-        if isinstance(rep_delay, list):
-            rep_delay = rep_delay[0]
-        rep_delay = rep_delay * 1e6  # convert sec to μs
+            rep_delay = rep_delay * 1e6  # convert sec to μs
+    else:
+        rep_delay = None
```

Comment:
It should be fine to still input this in the Qobj. For example in softtesting this feature on devices.

Reply:
What is the suggested action?

Reply:
Addressed in https://github.com/Qiskit/qiskit-terra/pull/4830/commits/4112f18a05921dd508953e6ca6aae28e078ea550

Reply:
If the value is None I suppose it shouldn't be added to the run_config.

Reply:
Yes that is what happens. https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/compiler/assemble.py#L340.